src/util.js

```diff
@@ -3,3 +3,27 @@ export function selectLast(node, selector) {
 	let { length } = nodes;
 	return length ? nodes[length - 1] : null;
 }
+
+function overflowBottom(node) {
+	let parentDimensions = node.parentElement.getBoundingClientRect();
+	let nodeDimensions = node.getBoundingClientRect();
+
+	return (nodeDimensions.top + nodeDimensions.height) > (parentDimensions.top + parentDimensions.height);
+}
+
+function overflowTop(node) {
+	let parentDimensions = node.parentElement.getBoundingClientRect();
+	let nodeDimensions = node.getBoundingClientRect();
+
+	return nodeDimensions.top < parentDimensions.top;
+}
+
+// NB: only supports vertical scrolling
+export function scrollIfNecessary(node) {
+	if (overflowBottom(node)) {
+		node.scrollIntoView(false);
+	}
+	if (overflowTop(node)) {
+		node.scrollIntoView(true);
+	}
```

Comment:
I'd just inline those two helper functions: They're not used anywhere else and we don't want to do the same work twice (namely invoking `#getBoundingClientRect`).

```suggestion
// NB: only supports vertical scrolling
export function scrollIfNecessary(node) {
	let parentDimensions = node.parentElement.getBoundingClientRect();
	let nodeDimensions = node.getBoundingClientRect();

	let top1 = nodeDimensions.top;
	let top2 = parentDimensions.top;
	if(top1 < top2) {
		node.scrollIntoView(true);
	} else if((top1 + nodeDimensions.height) > (top2 + parentDimensions.height)) {
		node.scrollIntoView(false);
	}
```